data-avro/src/main/java/com/linkedin/data/avro/DataTranslator.java

```diff
@@ -355,7 +355,9 @@ private Object translate(Object value, DataSchema dataSchema, Schema avroSchema)
             boolean isOptional = field.getOptional();
             DataSchema fieldDataSchema = field.getType();
             Schema fieldAvroSchema = avroSchema.getField(fieldName).schema();
-            if (isOptional && (fieldDataSchema.getDereferencedType() != DataSchema.Type.UNION))
+            if (isOptional && (fieldDataSchema.getDereferencedType() != DataSchema.Type.UNION) ||
```

Comment:
There are a few things wrong with this if-statement:
1. `isOptional` is no longer necessary, since we can no longer assume that this scenario is unique to optional fields.
2. It would be a lot simpler if you just made this consistent with the other if-statement in this diff; this condition should simply see if the pegasus type is *not* a union yet the Avro type *is* a union.
3. It has a lot of unnecessary parentheses, though if you follow my suggestions above, this should solve itself.